File: Degradation/DynamicLinkLibraries/DataPerformer/DataPerformer.Interfaces/IDifferentialEquationSolver.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace DataPerformer.Interfaces
{
    /// <summary>
    /// Solver of differential equations system
    /// </summary>
    public interface IDifferentialEquationSolver : IMeasurements, IStateDoubleVariables
    {
        /// <summary>
        /// Calculates derivations
        /// </summary>
        void CalculateDerivations();

        /// <summary>
        /// Copies variables from processor to solver 
        /// </summary>
        /// <param name="offset">Offset</param>
        /// <param name="variables">Vector of all desktop differential equations variables</param>
        void CopyVariablesToSolver(int offset, double[] variables);

     }
}

File: src/TypeScript/TypeScriptLibrary/Library/Measurements/DifferentialEquations/Interfaces/IDifferentialEquationSolver.ts
import { ITimeMeasurementProvider } from "../../Interfaces/ITimeMeasurementProvider";

export interface IDifferentialEquationSolver
{
    /// <summary>
    /// Calculates derivations
    /// </summary>
   calculateDerivations(): void;

    /// <summary>
    /// Copies variables from processor to solver 
    /// </summary>
    /// <param name="offset">Offset</param>
    /// <param name="variables">Vector of all desktop differential equations variables</param>
    copyVariablesToSolver(offset: number, variables: number[]): void;

    ///
    /// Sets time provider
    ///
    setDifferentialEquationSolverTimeProvider(time: ITimeMeasurementProvider)  : void;

    //gets timer provider
    getDifferentialEquationSolverTimeProvider(): ITimeMeasurementProvider;


}

